DEVELOPMENT.md

```diff
@@ -0,0 +1,473 @@
+
+
+Example:
+
+Given environment variable `FEATURE_AWESOME_SAUCE='true'`
+
+```js
+if (FEATURE_AWESOME_SAUCE === 'true') {
+  doSomething();
+}
+```
+
+results in
+
+```js
+doSomething();
+```
+
+.
+
+## Build Logs
+
+Build logs are streamed directly to the database in realtime so they are immediately viewable by customers in the UI. Every night, build logs for completed builds for the day are archived in an S3 bucket and deleted from the database. This prevents the database from growing unreasonably. The S3 bucket (with assistance from cloud.gov) is configured with a lifecycle policy that expires the logs after 180 days in accordance with [cloud.gov's policies on log retention](https://cloud.gov/docs/deployment/logs/#web-based-logs-with-historic-log-data).
+
+Lifecycle policy:
+```
+{
+  Bucket: <bucket>,
+  LifecycleConfiguration: {
+    Rules: [
+      {
+        Expiration: {
+          Days: 180
+        },
+        ID: "Archive all objects 180 days after creation"
+        Status: "Enabled",
+      }
+    ]
+  }
+}
+```
+
+## Commits, Releases, and Deployment
+
+All commit messages on the default branch should follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification. In addition, an associated ticket number should be added where relevant. Examples:
+
+```
+feat: Add fail stuck builds queue and breakout timeout builds queue (#4138)
+fix: Remove spurious menu button from paginated admin views. (#4152)
+chore: Remove deprecated federalistUsersHelpers function (#4131)
+```
+
+This convention is enforced via a number of GitHub functions:
+- [This ruleset](https://github.com/cloud-gov/pages-core/settings/rules/26912) checks the commit message against a regex rule before it can be merged to the default branch.
+- No [merge commit pull requests](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) are allowed to the repo
+- The default branch enforces a [linear commit history](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#require-linear-history)
+
+The benefit of adhering to this convention is that we can more easily reason about our commit history and also generate nice changelogs via [Release Please](https://github.com/googleapis/release-please).
+
+If you didn't follow this convention while making your commits locally or on a development, you'll still have (at least) two opportunities to edit the commit history to match the Convention Commits specification:
+- while the code is on a non-default branch, you can perform an [interative rebase](https://git-scm.com/docs/git-rebase) to rewrite the history.
+- prior to merging a PR, you can select either "Squash and merge" or "Rebase and merge" which will change the commit message(s) in the PR.
```

Comment:
It **is** confusing that both of these are characterized as merges whereas merge commits on the feature branch are not allowed. This isn't particularly friendly, but I'm not sure how to make it better.